test/e2e_node/conformance_test.go

```diff
@@ -0,0 +1,116 @@
+	var cl *client.Client
+	var terminateCase ConformanceContainer
+	var invalidImageCase ConformanceContainer
+
+	BeforeEach(func() {
+		// Setup the apiserver client
+		cl = client.NewOrDie(&client.Config{Host: *apiServerAddress})
+		terminateCase = ConformanceContainer{
+			Container: api.Container{
+				Image:           "gcc.io/google_testcontainers/busybox",
+				Name:            "busybox",
+				Command:         []string{"echo", "'Hello World'"},
+				ImagePullPolicy: api.PullIfNotPresent,
+			},
+			//terminated soon.
+			Status: "terminated",
+			Client: cl,
+		}
+		invalidImageCase = ConformanceContainer{
+			Container: api.Container{
+				Image:           "foo.com/foo/foo",
+				Name:            "foo",
+				Command:         []string{"foo", "'Should not work'"},
+				ImagePullPolicy: api.PullIfNotPresent,
+			},
+			//waiting for image pull
+			Status: "waiting",
+			Client: cl,
+		}
+	})
+
+	Describe("container conformance test", func() {
+		Context("when running a container that terminates", func() {
+			It("it should start successfully [Conformance]", func() {
+				if err := terminateCase.Create(); err != nil && !errors.IsAlreadyExists(err) {
+					Expect(err).NotTo(HaveOccurred(), err.Error())
+					return
+				}
+
+				errs := Retry(time.Minute*1, time.Second*30, func() error {
+					_, err := terminateCase.IsStarted()
+					return err
+				})
+				Expect(errs).To(BeEmpty(), fmt.Sprintf("Failed to start container"))
+			})
+			It("it should report its status as 'terminated' [Conformance]", func() {
+				if container, cstatus, err := terminateCase.Get(); err != nil {
+					Expect(err).NotTo(HaveOccurred(), err.Error())
+				} else if err := terminateCase.Equal(container, cstatus); err != nil {
+					Expect(err).NotTo(HaveOccurred(), err.Error())
+				}
+			})
+			It("it should be possible to delete [Conformance]", func() {
+				if err := terminateCase.Delete(); err != nil {
+					Expect(err).NotTo(HaveOccurred(), err.Error())
+				}
+			})
+		})
+		Context("when running a container with an invalid image", func() {
+			It("it should not start successfully [Conformance]", func() {
+				if err := invalidImageCase.Create(); err != nil && !errors.IsAlreadyExists(err) {
+					Expect(err).NotTo(HaveOccurred(), err.Error())
+					return
+				}
+
+				errs := Retry(time.Minute*1, time.Second*30, func() error {
+					_, err := invalidImageCase.IsStarted()
+					return err
+				})
+				Expect(errs).NotTo(BeEmpty(), fmt.Sprintf("Container with invalid image should not start"))
+			})
+			It("it should report its status as 'waiting' [Conformance]", func() {
+				if container, cstatus, err := invalidImageCase.Get(); err != nil {
+					Expect(err).NotTo(HaveOccurred(), err.Error())
+				} else if err := invalidImageCase.Equal(container, cstatus); err != nil {
+					Expect(err).NotTo(HaveOccurred(), err.Error())
+				}
+			})
+			It("it should be possible to delete [Conformance]", func() {
+				if err := invalidImageCase.Delete(); err != nil {
```

Comment:
Why not just 
>
err = invalidImageCase.Delete()
Expect(err).NotTo(HaveOccurred())